lms/djangoapps/courseware/views/views.py

```diff
@@ -737,6 +737,13 @@ def _progress(request, course_key, student_id):
         and certs_api.cert_generation_enabled(course_key)
     )
 
+    # If current certificate is invalidated by instructor
+    # then don't show the generate button.
+    is_invalid = False
+    if show_generate_cert_btn and certs_api.is_certificate_invalid(student, course_key):
```

Comment:
I am calling `certs_api.is_certificate_invalid` only when `show_generate_cert_btn` is True. Second I needs the value of `certs_api.is_certificate_invalid` also for progress page.  The problem with following code is `is_certificate_invalid` will be called each time student access the page.  What do u think ?

```
is_invalid = certs_api.is_certificate_invalid(student, course_key)
show_generate_cert_btn = (
        is_active and CourseMode.is_eligible_for_certificate(enrollment_mode)
        and certs_api.cert_generation_enabled(course_key)
        and not is_invalid
    )
```